atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -219,40 +221,101 @@ def _get_remote_urls_helper(self, koji_build):
-                remote_source['dest'] = '-'.join([koji_build['nvr'], archive['filename']])
-                dest_name = remote_source['dest']
-                remote_sources_urls.append(remote_source)
-
-            elif archive['type_name'] == 'json' and archive['filename'] == 'remote-source.json':
-                cachito_json_url = os.path.join(remote_sources_path, archive['filename'])
-
-        # we will add entry to remote_json_map only when
-        # build has remote sources archive and json
-        # it will be used later to get correspondent remote-source.json for each remote sources
-        # archive from all parents
-        if cachito_json_url and dest_name:
-            remote_json_map = {dest_name: cachito_json_url}
-
-            if len(remote_sources_urls) > 1:
-                raise RuntimeError('There can be just one remote sources archive, got : {}'.
-                                   format(remote_sources_urls))
-
-        elif bool(cachito_json_url) != bool(dest_name):
-            raise RuntimeError('Remote sources archive or remote source json missing, '
-                               'remote source archive: {}, remote source json: {}'.
-                               format(dest_name, cachito_json_url))
+        if 'remote_source_url' in koji_build['extra']['image']:
+            self.log.debug('remote_source_url defined')
+            remote_source = {}
+            remote_source['url'] = os.path.join(remote_sources_path, REMOTE_SOURCES_FILENAME)
+            remote_source['dest'] = '-'.join([koji_build['nvr'], REMOTE_SOURCES_FILENAME])
+            remote_sources_urls.append(remote_source)
+            cachito_json_url = os.path.join(remote_sources_path, REMOTE_SOURCES_JSON_FILENAME)
+            remote_json_map[remote_source['dest']] = cachito_json_url
+
+            archive_found = False
+            json_found = False
+            all_archives = []
+
+            for archive in archives:
+                if archive['filename'] == REMOTE_SOURCES_FILENAME:
+                    archive_found = True
+                elif archive['filename'] == REMOTE_SOURCES_JSON_FILENAME:
+                    json_found = True
+                all_archives.append(archive['filename'])
+
+            if not (archive_found and json_found):
+                message = ''
+                if not archive_found:
+                    message = 'Remote sources archive'
+                if not json_found:
+                    if message:
+                        message += ' and remote sources json'
+                    else:
+                        message = 'Remote sources json'
+                message += ' missing.'
+                raise RuntimeError(message)
+
+            elif len(archives) > 2:
+                raise RuntimeError('There can be just one remote sources archive and one '
+                                   'remote sources json, got: {}'.format(all_archives))
```

Comment:
Can be this extracted into separate method for better readability?

Reply:
ok